### PR TITLE
Fix viewing branch list when a branch has no associated platforms

### PR DESCRIPTION
**Pull request by hbiede**
# Checklist

- [X] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.

Branch list would error when the following command was performed: `eas branch:create "test" && eas branch:list`

# How

How did you build this feature or fix this bug and why?

Fix error where platform-less branches would throw a type error

# Test Plan

Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.

Run `eas branch:create "test" && eas branch:list` with `v1.2.0`, see an error, run again with this branch and see it succeed

### CODE REVIEW

File: packages/eas-cli/src/branch/queries.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ function renderPageOfBranches(
         branch.updates[0]?.group ?? 'N/A',
         getPlatformsForGroup({
           updates: branch.updates,
-          group: branch.updates[0]?.group,
+          group: branch.updates[0]?.group ?? [],
```
Comment (wschurman) [inline packages/eas-cli/src/branch/queries.ts:99]: I think this will have an incorrect type? The type of the group property is string, so this would pass an array.

I played around with this locally and think the best solution is to explicitly declare the result of `branch.updates[0]` to be typed as potentially undefined.

```typescript
  table.push(
    ...currentPage.map(branch => {
      const update = branch.updates[0] as UpdateBranchFragment['updates'][0] | undefined;
      return [
        branch.name,
        formatUpdate(update),
        update?.runtimeVersion ?? 'N/A',
        update?.group ?? 'N/A',
        update?.group
          ? getPlatformsForGroup({
              updates: branch.updates,
              group: update.group,
            })
          : 'N/A',
      ];
    })
  );
```

File: packages/eas-cli/src/update/utils.ts [hunk 2]
```diff
@@ -47,14 +47,14 @@ export function getPlatformsForGroup({
 }
 
 export function formatPlatformForUpdateGroup(
-  updateGroup: {
+  updateGroup?: {
```
Comment (wschurman) [inline packages/eas-cli/src/update/utils.ts:50]: then, with the changes above, you should be able to revert the changes to this file

**Review by wschurman — CHANGES_REQUESTED** [review]
Nice find! @kgc00 should also take a look.

File: packages/eas-cli/src/branch/queries.ts (hunk 1 above)
Comment (hbiede) [reply packages/eas-cli/src/branch/queries.ts:99]: Yep, I noticed that, fixed it in a different way, committed it, and forgot to push 🤦🏼

File: packages/eas-cli/src/update/utils.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export function formatPlatformForUpdateGroup(
     platform: string;
   }[]
 ): string {
-  return updateGroup.length === 0
+  return updateGroup!.length === 0
```
Comment (wschurman) [inline packages/eas-cli/src/update/utils.ts:55]:
```suggestion
  return updateGroup.length === 0
```

File: packages/eas-cli/src/branch/queries.ts [hunk 4]
```diff
@@ -89,16 +89,21 @@ function renderPageOfBranches(
...
+        const update: UpdateBranchFragment['updates'][0] | undefined = branch.updates[0];
+        return [
+          branch.name,
+          formatUpdate(update),
+          update?.runtimeVersion ?? 'N/A',
+          update?.group ?? 'N/A',
+          update?.group
+            ? getPlatformsForGroup({
+                updates: branch.updates,
+                group: update?.group ?? '',
```
Comment (wschurman) [inline packages/eas-cli/src/branch/queries.ts:102]:
```suggestion
                group: update.group,
```

**Review by wschurman — APPROVED** [review]
Just a few more changes before commit, but looks good to me otherwise.

**Review by kgc00 — APPROVED** [review]
Looks good to me! Thanks for submitting. Fwiw, I think `getPlatformsForGroup` should be updated to handle the `undefined` case (and can roll that into a PR I'm working on).

**Review by wkozyra95 — APPROVED** [review]
